ocaml/Makefile

```diff
@@ -0,0 +1,171 @@
+ifeq ($(UNAME),Darwin)
+  SO		= dylib
+  OCAML_SO	= so
+else ifeq ($(UNAME),Linux)
+  SO 		= so
+  OCAML_SO	= so
+  CFLAGS	+= -fPIC
+else ifeq ($(OS),Windows_NT)
+  $(error "Windows is not supported at the moment.")
+else ifeq ($(UNAME),FreeBSD)
+  SO 		= so
+  OCAML_SO	= so
+  CFLAGS	+= -fPIC
+endif
+
+# Config
+# TODO: Read this from the main config.
+STATIC_C_LIB_NAME=hacl_static
+DYNAMIC_C_LIB_NAME=hacl
+BLOCKLIST=Hacl_Bignum256_32.c  Hacl_Bignum32.c Hacl_Bignum4096_32.c \
+			Hacl_Bignum4096.c Hacl_Bignum.c Hacl_Bignum64.c Hacl_Bignum_Base.c \
+			Hacl_GenericField32.c Hacl_GenericField64.c \
+			Hacl_HPKE_Curve64_CP128_SHA256.c Hacl_HPKE_Curve64_CP128_SHA512.c \
+			Hacl_HPKE_Curve64_CP256_SHA256.c Hacl_HPKE_Curve64_CP256_SHA512.c \
+			Hacl_HPKE_Curve64_CP32_SHA256.c Hacl_HPKE_Curve64_CP32_SHA512.c \
+			Hacl_HPKE_Curve51_CP256_SHA256.c Hacl_HPKE_Curve51_CP256_SHA512.c \
+			Hacl_HPKE_Curve64_CP256_SHA256.c Hacl_HPKE_Curve64_CP256_SHA512.c \
+			Hacl_HPKE_P256_CP256_SHA256.c
+ifeq (,$(TOOLCHAIN_CAN_COMPILE_VEC128))
+BLOCKLIST+=Hacl_Chacha20Poly1305_128.c Hacl_Poly1305_128.c \
+			Hacl_Hash_Blake2s_128.c Hacl_Streaming_Blake2s_128.c
+endif
+ifeq (,$(TOOLCHAIN_CAN_COMPILE_VEC256))
+BLOCKLIST+=Hacl_Bignum256.c Hacl_Chacha20Poly1305_256.c Hacl_Chacha20_Vec256.c \
+			Hacl_HKDF_Blake2b_256.c Hacl_HMAC_Blake2b_256.c \
+			Hacl_Hash_Blake2b_256.c Hacl_Poly1305_256.c Hacl_SHA2_Vec256.c \
+			Hacl_Streaming_Blake2b_256.c Hacl_Streaming_Poly1305_256.c
+endif
+ifeq (,$(TOOLCHAIN_CAN_COMPILE_VALE))
+BLOCKLIST+=Hacl_Curve25519_64.c evercrypt_vale_stubs.c EverCrypt_Vale.c
+endif
+C_PATH?=c
+C_LIB_PATH?=$(C_PATH)/lib/
+C_LIB?=$(C_LIB_PATH)lib$(STATIC_C_LIB_NAME).a
+C_DYN_LIB?=$(C_LIB_PATH)lib$(DYNAMIC_C_LIB_NAME).$(SO)
+C_INCLUDES?=-Ic/include/hacl/ \
+			-Ic/include/vale/ \
+			-Ic/include/ \
+			-I../include/ # XXX: internal is used in lib/
+KARAMEL_INCLUDES?=-Ic/include/kremlin/include \
+				  -Ic/include/kremlin/kremlib/dist/minimal/
+
+OCAMLOPT=ocamlfind opt -package ctypes,ctypes.stubs -linkpkg -I lib
+OCAMLDEP=ocamlfind dep -I lib -slash
+
+OCAMLC=ocamlfind c -g -package ctypes,ctypes.stubs -linkpkg -I lib
+
+CFLAGS += -I "$(shell ocamlfind query ctypes)" -I "$(shell ocamlfind c -where)" \
+		  $(C_INCLUDES) $(KARAMEL_INCLUDES)
+
+# Don't include bindings for files that cannot be compiled.
+BLOCKLIST_ML=$(patsubst %.c,%,$(BLOCKLIST))
+ALL_OCAML=$(filter-out $(BLOCKLIST_ML),$(patsubst lib_gen/%_gen.ml,%,$(wildcard lib_gen/*_gen.ml)))
+
+# File names.
+ALL_BINDINGS=$(patsubst %,lib/%_bindings.cmx,$(ALL_OCAML))
+ALL_GENERATORS=$(patsubst %,lib_gen/%_gen.exe, $(ALL_OCAML))
+ALL_ML_STUBS=$(patsubst %,lib/%_stubs.cmx,$(ALL_OCAML))
+ALL_C_STUBS=$(patsubst %,lib/%_c_stubs.o,$(ALL_OCAML))
+
+include .depend.ocaml
+include ctypes.depend
+
+# - copy and clean config.h
+setup:
+ifeq ($(wildcard $(C_PATH)),)
+		mkdir $(C_PATH)
+		cp -r ../build/installed/* $(C_PATH)/
+		cp $(C_LIB) .
+		sed -ri '/(\/\/.*)|(\/\*.*\*\/)/d' ./c/include/hacl/config.h
```

Comment:
does this sed invocation work on OSX?

Reply:
on OSX by default I believe `sed` is POSIX `sed` which does not accept `-i` with no argument...

Reply:
Works for me. I expect anyone using something like make to use the GNU tools :)